src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+else()
   target_link_libraries(benchmark ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
i'm pretty sure this will statically link pthread into the static library, no? i'm not completely familiar with how target_link_libraries works under the different variants in cmake.